### PR TITLE
[15.07] Release notes for 15.07.

### PR DESCRIPTION
**Pull request by jmchilton**
This includes a script to boot strap the notes for future releases, updated announcement page for 15.07, and new announcement page for 15.10.

Few points to consider during review:
- Feel free to push smaller stuff into my branch directly (open a PR if you don't have access yet, I'll add you).
- As always the ordering is somewhat arbitrary and reflects a John worldview of Galaxy, please start a conversation if you think other items should be highlighted instead.
- Please review 15.10 release schedule and comment (@natefoo in particular).
- The script used to bootstrap the release notes includes a list of github users that don't get explicit credit for their work (essentially old devteam). Depending on how you look at this - it could be seen as a special privilege for old devteam or for the newer committers. My thought on how to handle this is just let each person decide if they want credit or not in the release notes by modifying that list.

### CODE REVIEW

File: doc/source/releases/15.07_announce.rst [hunk 1]
```diff
@@ -2,22 +2,60 @@
...
+  The interactive environments (IE) framework features several updates this release
+  and Galaxy is now distributed with a new IE for RStudio implemented by Eric Rasche.
+
+**Workflow Editor Enhancements**
+  The workflow editor has been significantly revamped - it now uses newer tool form
+  components and allow construction of workflows with tools explicitly producing
+  output collections.
 
-Schedule
+**Reports Application Enhancements**
```
Comment (martenson) [inline doc/source/releases/15.07_announce.rst]: I think this is not in 15.07

File: doc/source/releases/15.10_announce.rst [hunk 2]
```diff
@@ -0,0 +1,9 @@
+===========================================================
+October 2015 Galaxy Release (v 15.10)
+===========================================================
+
+
+Schedule
+===========================================================
+ * Planned Freeze Date: October 5, 2015
+ * Planned Release Date: October 26, 2015
```
Comment (natefoo) [inline doc/source/releases/15.10_announce.rst:9]: +1 on the schedule for 15.10

File: doc/source/releases/15.07_announce.rst (hunk 1 above)
Comment (jmchilton) [inline doc/source/releases/15.07_announce.rst]: Really? Ugh - is there a problem with my bootstrap script :(. I'll look into this.

File: doc/source/releases/15.10_announce.rst [hunk 3]
```diff
@@ -0,0 +1,9 @@
+===========================================================
+October 2015 Galaxy Release (v 15.10)
```
Comment (martenson) [inline doc/source/releases/15.10_announce.rst:2]: the planned release was only in `dev` last time, if we put it in the `release_15.07' the latest actual release will never be on top

File: doc/source/releases/15.07_announce.rst (hunk 1 above)
Comment (jmchilton) [inline doc/source/releases/15.07_announce.rst]: Yup - I don't understand what went wrong. Maybe there was some time between when Nate added the commit that created the branch and when it was merged back into dev.

**Comment by martenson** [comment]
@jmchilton Do you want us to modify this or are you working on the script-generator?

**Comment by jmchilton** [comment]
Going through and modifying by hand now. My WIP:

```
Deleting -

X 370,
X 410,
X 454,
X 462,
X 463,
X 472 
X 473
X 472-488
X 491-530,

X 532, 533, 536-546, 550, 553, 561, 565, 568, 569, 575, 577, 581-594

Adding - 528, 639, 649
```

**Comment by jmchilton** [comment]
Okay - @martenson talked about the announcement not just being in dev and he is okay with that. I've fixed the release notes (replaced reports highlight with committers policy change). Removed the WIP tag.

File: doc/source/releases/15.07_announce.rst [hunk 4]
```diff
@@ -2,22 +2,62 @@
...
+
+**Workflow Editor Enhancements**
+  The workflow editor has been significantly revamped - it now uses newer tool form
+  components and allow construction of workflows with tools explicitly producing
+  output collections.
+
+**Policies for Committers and Pull Requests**
+  The process for adding committers to the project and fielding pull requests has 
+  been formalized and documented in the source code. Three new committers have
+  added to the project - Björn Grüning, Nicola Soranzo, and Eric Rasche.
```
Comment (nsoranzo) [inline doc/source/releases/15.07_announce.rst:23]: "have added" -> "have been added" ?

**Comment by natefoo** [comment]
:+1:

**Comment by dannon** [comment]
+1

**Pull request merged**